netty/src/test/java/io/grpc/netty/NettyClientHandlerTest.java

```diff
@@ -457,6 +457,171 @@ public void ping_failsWhenChannelCloses() throws Exception {
   }
 
   @Test
+  public void dataPingSentOnDataRecieved() throws Exception {
+    createStream();
+    handler().setAutoTuneFlowControl(true);
+
+    channelRead(dataFrame(3, false));
+
+    assertEquals(1, handler().flowControlPinger().getPingCount());
+  }
+
+  @Test
+  public void dataPingAckIsRecognized() throws Exception {
+    createStream();
+    // handler().flowControlPinger().setTesting();
+
+    channelRead(dataFrame(3, false));
+    long pingdata = handler().flowControlPinger().payload();
+    ByteBuf buffer = handler().ctx().alloc().buffer(8);
+    buffer.writeLong(pingdata);
+    channelRead(pingFrame(true, buffer));
+
+    assertEquals(1, handler().flowControlPinger().getPingCount());
+    assertEquals(1, handler().flowControlPinger().getPingReturn());
+  }
+
+  @Test
+  public void dataSizeSincePingAccumulates() throws Exception {
```

Comment:
Since this and some other tests appear identical between client/server, can they be moved to `NettyHandlerTestBase`?